Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -851,116 +844,71 @@ public override int GetHashCode()
         /// <summary>
         /// This should be run after deserialization of a PeptideWithSetModifications, in order to set its Protein and Modification objects, which were not serialized
         /// </summary>
-        public void SetNonSerializedPeptideInfo(Dictionary<string, Modification> idToMod, Dictionary<string, Protein> accessionToProtein)
+        public void SetNonSerializedPeptideInfo(Dictionary<string, Modification> idToMod, Dictionary<string, Protein> accessionToProtein, DigestionParams dp)
```

Comment:
are the digestionparams known after deserialization? i.e., where does the "dp" parameter come from?

Reply:
The dp is known from the search task that's being done. Currently, each digestion parameter needs its own index, such that all peptides within a given index contain the same digestion parameter object. This is output in the index params file.